tests/unit_tests/sql/transpile_to_dialect_test.py

```diff
@@ -0,0 +1,271 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Tests for transpile_to_dialect function in superset/sql/parse.py
+"""
+
+import pytest
+
+from superset.sql.parse import transpile_to_dialect
+
+
+def test_postgres_single_quote_escaping():
+    """PostgreSQL keeps SQL-92 standard: double single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "postgresql")
+    assert "O''Hara" in result
+    assert "name" in result
+
+
+def test_databricks_single_quote_escaping():
+    """Databricks uses backslash escaping for single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "databricks")
+    assert "name" in result
+    # Databricks uses backslash escaping
+    assert "O\\'Hara" in result or "O''Hara" in result
+
+
+def test_mysql_single_quote_escaping():
+    """MySQL uses backslash escaping for single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "mysql")
+    assert "name" in result
+    # MySQL may use backslash or double-quote escaping
+    assert (
+        "Connor" in result.replace("O''Hara", "Connor").replace("O\\'Hara", "Connor")
+        or "Hara" in result
+    )
+
+
+def test_sqlite_single_quote_escaping():
+    """SQLite uses SQL-92 standard: double single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "sqlite")
+    assert "O''Hara" in result
+    assert "name" in result
+
+
+def test_snowflake_single_quote_escaping():
+    """Snowflake handles single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "snowflake")
+    assert "name" in result
+    assert "Hara" in result
+
+
+def test_bigquery_single_quote_escaping():
+    """BigQuery handles single quotes."""
+    sql = "name = 'O''Hara'"
+    result = transpile_to_dialect(sql, "bigquery")
+    assert "name" in result
+    assert "Hara" in result
+
+
+def test_compound_filter_with_quotes():
+    """Test compound filters with quoted strings."""
+    sql = "(name = 'O''Hara' AND status = 'active')"
+    result = transpile_to_dialect(sql, "postgresql")
+    assert "Hara" in result
+    assert "active" in result
+    assert "AND" in result
```

Comment:
It's better to test for the behavior we want... if `result` was `Hara active AND` this test would pass, even though the result is wrong. So let's do something like this here and in all tests in this file:

```suggestion
    assert transpile_to_dialect(sql, "postgresql") == "(name = 'O''Hara' AND status = 'active')"
```

That is, check the actual generated string, not just if substrings are present.